remoteworkitem/trackerquery_repository_test.go

```diff
@@ -50,54 +62,62 @@ func TestTrackerQuerySave(t *testing.T) {
+		req := &http.Request{Host: "localhost"}
+		params := url.Values{}
+		ctx := goa.NewContext(context.Background(), nil, req, params)
+
+		err := queryRepo.Delete(ctx, "asdf")
 		assert.IsType(t, NotFoundError{}, err)
 
-		tracker, _ := trackerRepo.Create(context.Background(), "http://api.github.com", ProviderGithub)
-		tq, _ := queryRepo.Create(context.Background(), "is:open is:issue user:arquillian author:aslakknutsen", "15 * * * * *", tracker.ID)
-		err = queryRepo.Delete(context.Background(), tq.ID)
+		tracker, _ := trackerRepo.Create(ctx, "http://api.github.com", ProviderGithub)
+		tq, _ := queryRepo.Create(ctx, "is:open is:issue user:arquillian author:aslakknutsen", "15 * * * * *", tracker.ID, space.SystemSpace)
+		err = queryRepo.Delete(ctx, tq.ID)
 		assert.Nil(t, err)
 
-		tq, err = queryRepo.Load(context.Background(), tq.ID)
+		tq, err = queryRepo.Load(ctx, tq.ID)
 		assert.IsType(t, NotFoundError{}, err)
 		assert.Nil(t, tq)
 
-		tq, err = queryRepo.Load(context.Background(), "100000")
+		tq, err = queryRepo.Load(ctx, "100000")
 		assert.IsType(t, NotFoundError{}, err)
 		assert.Nil(t, tq)
 	})
 }
 
 func TestTrackerQueryList(t *testing.T) {
 	doWithTrackerRepositories(t, func(trackerRepo application.TrackerRepository, queryRepo application.TrackerQueryRepository) {
-		trackerqueries1, _ := queryRepo.List(context.Background())
+		req := &http.Request{Host: "localhost"}
+		params := url.Values{}
+		ctx := goa.NewContext(context.Background(), nil, req, params)
+
+		trackerqueries1, _ := queryRepo.List(ctx)
 
-		tracker1, _ := trackerRepo.Create(context.Background(), "http://api.github.com", ProviderGithub)
-		queryRepo.Create(context.Background(), "is:open is:issue user:arquillian author:aslakknutsen", "15 * * * * *", tracker1.ID)
-		queryRepo.Create(context.Background(), "is:close is:issue user:arquillian author:aslakknutsen", "15 * * * * *", tracker1.ID)
+		tracker1, _ := trackerRepo.Create(ctx, "http://api.github.com", ProviderGithub)
+		queryRepo.Create(ctx, "is:open is:issue user:arquillian author:aslakknutsen", "15 * * * * *", tracker1.ID, space.SystemSpace)
+		queryRepo.Create(ctx, "is:close is:issue user:arquillian author:aslakknutsen", "15 * * * * *", tracker1.ID, space.SystemSpace)
 
-		tracker2, _ := trackerRepo.Create(context.Background(), "http://issues.jboss.com", ProviderJira)
-		queryRepo.Create(context.Background(), "project = ARQ AND text ~ 'arquillian'", "15 * * * * *", tracker2.ID)
-		queryRepo.Create(context.Background(), "project = ARQ AND text ~ 'javadoc'", "15 * * * * *", tracker2.ID)
+		tracker2, _ := trackerRepo.Create(ctx, "http://issues.jboss.com", ProviderJira)
+		queryRepo.Create(ctx, "project = ARQ AND text ~ 'arquillian'", "15 * * * * *", tracker2.ID, space.SystemSpace)
+		queryRepo.Create(ctx, "project = ARQ AND text ~ 'javadoc'", "15 * * * * *", tracker2.ID, space.SystemSpace)
 
-		trackerqueries2, _ := queryRepo.List(context.Background())
+		trackerqueries2, _ := queryRepo.List(ctx)
```

Comment:
@kwk context.Backgroun is empty while ctx is initialized with some parameters that will later use.